### ISSUE TITLE
Splunk Logging Aggregator doesnt allow Splunk to use Supplied Timestamps

### ISSUE THREAD

**Issue by digitalbadger-uk** [ansible/awx#14236]
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx/issues) for duplicates.
- [X] I understand that AWX is open source software provided for free and that I might not receive a timely response.
- [X] I am **NOT** reporting a (potential) security vulnerability. (These should be emailed to `security@ansible.com` instead.)

### Bug Summary

When configuring the logging aggregator to use Splunk, the log output is generated in a Splunk event format with the json log entry (including @timestamp field) being held within an event like: {"event": {<log detail>}
From a Splunk perspective this would usually be directed at the Splunk HEC which then processes is.  Being in the event format the appropriate url to receive the message would be https://splunk:8088/services/collector/event - however when using this url splunk doesnt look for timestamps in the event and will tag the message with the time it was received.  In order for Splunk to extract the message timestamp when using the event url it needs to have the auto_extract_timestamp=true parameter set for the url - so https://splunk/services/collector/event?auto_extract_timestamp=true 
When the LOG_AGGREGATOR_HOST is set to https://splunk/services/collector/event?auto_extract_timestamp=true  the parameter seems to have been url encoded (/services/collector/event?auto_extract_timestamp%3Dtrue) and is not parsed as a parameter by Splunk.
So either the LOG_AGGREGATOR_HOST needs to be able to accept the auto_extract_timestamp parameter to be passed to the Splunk url or take the json log entry out of the event wrapper and then the splunk /services/collector/raw url can be used which automatically scans for timestamps.

### AWX version

22.5.0

### Select the relevant components

- [ ] UI
- [ ] UI (tech preview)
- [X] API
- [ ] Docs
- [ ] Collection
- [ ] CLI
- [X] Other

### Installation method

docker development environment

### Modifications

no

### Steps to reproduce

Configure logging aggregator to send to Splunk using URL "https://splunk:8080/services/collector/event?auto_extract_timestamp=true"
Configure HEC event receiver on splunk

### Expected results

The timestamp on the splunk event should match the value in @timestamp field of the logged event

### Actual results

Splunk shows the received time as the event time not the time from @timestamp
![image](https://github.com/ansible/awx/assets/63948954/d7629d70-b68a-45ba-8629-dc4e095e0c27)

**Comment by digitalbadger-uk** [comment]
Ok so it looks like the issue is in the formatters.py with the following lines:
        if settings.LOG_AGGREGATOR_TYPE == 'splunk':
            # splunk messages must have a top level "event" key
            message = {'event': message}
This wraps the log entry that is being sent int an "event" but without properly considering the rest of Splunks requirements.  
Removing these lines would allow the event to be successfully received using the raw receiver (https://splunk:8088/services/collector/raw) - however I guess removing these lines may then cause an issue for anyone with their environment configured to receive their logging via the event receiver (https://splunk:8088/services/collector/event) which would reject anything without the wrapper.  
But when sending the logs to spunk using the event receiver Splunk wont parse the received event for a timestamp unless the parameter auto_extract_timestamp is correctly set in the url, or a time field is provided before the event wrapper with the time defined using epoch time.  Adding this time field might be the easier option and would actually fix the time mismatch issues for anyone already using the event receiver.

**Comment by fosterseth** [comment]
> Adding this time field might be the easier option and would actually fix the time mismatch issues for anyone already using the event receiver

would you be interested in getting a patch up for this change, and testing whether it works?

or, would it be better to expose a setting to allow raw vs event logging in splunk (and add 'event' key depending on this setting)

**Comment by digitalbadger-uk** [comment]
Yep it ended up working quite nicely.  I submitted https://github.com/ansible/awx/pull/14246 for it.
Thanks
